lib/Statistics/GATracking.php

```diff
@@ -58,7 +54,9 @@ private function addParams($extracted_links, $ga_campaign, $internal_host = null
         // Process only internal links (i.e. pointing to current site)
         continue;
       }
-      $processed_link = WPFunctions::get()->addQueryArg($params, $extracted_link['link']);
+      list($path, $search, $hash) = $this->splitLink($extracted_link['link']);
+      $search = empty($search) ? $params : $search . '&' . $params;
+      $processed_link = $path . '?' . $search . '#' . $hash;
```

Comment:
I'm afraid this will add `#` at the end of every link without `$hash`. Could you handle that?